src/jit/codegenarm64.cpp

```diff
@@ -3409,11 +3411,38 @@ void CodeGen::genCodeForStoreInd(GenTreeStoreInd* tree)
 
         if (tree->gtFlags & GTF_IND_VOLATILE)
         {
-            // issue a full memory barrier a before volatile StInd
-            instGen_MemoryBarrier();
+            bool useStoreRelease = (dataReg >= REG_INT_FIRST) && (dataReg <= REG_ZR) && !addr->isContained() &&
+                                   ((attr == EA_1BYTE) || !(tree->gtFlags & GTF_IND_UNALIGNED));
```

Comment:
It also should not be possible to have the case of an "unaligned" 1 byte load/store.
A check could be added when we decide to set that flag.